slemicro/make_unattended.sh

```
@@ -67,91 +67,113 @@ TMPDIR="$(mktemp -d -t make_unattended.XXXXXX)"
-
-# First step is to extract the initrd filesystem content (the second cpio)
-# pushd and popd commands are needed because lsinitrd doesn't have a way to extract the data than to the current folder
-pushd "${TMPDIR}"/init > /dev/null
-lsinitrd --unpack ./initrd
-popd > /dev/null
-
-# Then, in order to be able to recreate it, we need to get the first cpio as well
-# We just list the content from it but we need stderr as this is where cpio prints the number of blocks written
-# The usual cat initrd | cpio -it gives a pipefail (because of cat), so use -F instead (which should be better)
-BLOCKS=$(cpio -itF "${TMPDIR}"/init/initrd 2>&1 | awk '/blocks/ { print $1 }')
-dd if="${TMPDIR}"/init/initrd skip=0 count="${BLOCKS}" of="${TMPDIR}"/first >/dev/null 2>&1
-
-# We have everything we need from initrd
-rm -f "${TMPDIR}"/init/initrd
-
-# We inject the unattended flag
-echo "kiwi_oemunattended='true'" >> "${TMPDIR}"/init/.profile
+
+# First step is to extract the .profile from the second cpio
+lsinitrd "${TMPDIR}"/initrd .profile > "${TMPDIR}"/.profile
+
+# Then, append the kiwi_oem flags
+echo "kiwi_oemunattended='true'" >> "${TMPDIR}"/.profile
 
 # Specify the destination disk if present
 if [ -n "${WRITETO:-}" ]; then
-	echo "kiwi_oemunattended_id='${WRITETO}'" >> "${TMPDIR}"/init/.profile
+	echo "kiwi_oemunattended_id='${WRITETO}'" >> "${TMPDIR}"/.profile
 fi
 
-# This will generate the "second" cpio
-pushd "${TMPDIR}"/init/ > /dev/null
-find . | cpio -o -H newc -F "${TMPDIR}"/second 2>/dev/null
-popd > /dev/null
-
-# We can clean up the extracted one
-rm -Rf "${TMPDIR}"/init
-
-# Now we need to compress the filesystem
-zstd -q "${TMPDIR}"/second
-rm -f "${TMPDIR}"/second
-
-# And create the proper initrd by concatenating both
-cat "${TMPDIR}"/first "${TMPDIR}"/second.zst > "${TMPDIR}"/initrd
-rm -f "${TMPDIR}"/first "${TMPDIR}"/second.zst
+# Finally, we append the .profile into the initrd as:
+echo "${TMPDIR}"/.profile | cpio -o -F "${TMPDIR}"/initrd 2>/dev/null
```

I suggest to use `--quiet` instead of `2>/dev/null` to not hide error messages.

---

Good suggestion, thanks!